components/Review/CreateEditReview.tsx

```diff
@@ -0,0 +1,191 @@
+  Text,
+  StyleSheet,
+  Pressable,
+  SafeAreaView,
+  TextInput,
+} from "react-native";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { Entypo, Ionicons } from "@expo/vector-icons";
+import { useDispatch } from "react-redux";
+import { useAppSelector } from "../../redux/hooks";
+import { selectUser } from "../../redux/selectors/user";
+import { RatingInput } from "./RatingView";
+import ReviewAPI from "./ReviewAPI";
+import { addReview, editReview } from "../../redux/slices/review";
+import BizReviewHeader from "./BizReviewHeader";
+import { selectBusinessById } from "../../redux/selectors/business";
+import { RootStackParamList } from "../../route-settings";
+import { selectReviewById } from "../../redux/selectors/review";
+import { ReviewType } from "../../src/APITypes";
+
+const styles = StyleSheet.create({
+  back: {
+    flexDirection: "row",
+    alignItems: "center",
+    marginLeft: -3,
+  },
+  backText: {
+    fontFamily: "Mada-SemiBold",
+    fontSize: 18,
+    textAlign: "left",
+    padding: 10,
+    paddingLeft: "1%",
+  },
+  post: {
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  postText: {
+    fontFamily: "Mada-Black",
+    color: "#DA5125",
+    fontSize: 18,
+    textAlign: "right",
+    padding: 10,
+    paddingLeft: "1%",
+  },
+  descriptor: {
+    color: "grey",
+    fontFamily: "Mada-Medium",
+    paddingTop: 10,
+    paddingBottom: 8,
+  },
+});
+
+interface BackButtonProps {
+  action: () => void;
+}
+
+function BackButton({ action }: BackButtonProps) {
+  return (
+    <Pressable onPress={action}>
+      <View style={styles.back}>
+        <Entypo name="chevron-left" size={24} />
+        <Text style={styles.backText}>Back</Text>
+      </View>
+    </Pressable>
+  );
+}
+
+function PostButton({ action }: BackButtonProps) {
+  return (
+    <Pressable onPress={action}>
+      <View style={styles.post}>
+        <Text style={styles.postText}>Post</Text>
+      </View>
+    </Pressable>
+  );
+}
+
+function Line() {
+  return (
+    <View
+      style={{
+        borderBottomColor: "#D9D9D9",
+        borderBottomWidth: 1,
+        marginTop: 10,
+        marginRight: "-10%",
+        marginLeft: "-10%",
+      }}
+    />
+  );
+}
+
+function Margin() {
+  return <View style={{ flex: 1 }} />;
+}
+
+type CreateEditReviewProps = NativeStackScreenProps<
+  RootStackParamList,
+  "CreateEditReview"
+>;
+
+function CreateEditReview({ navigation, route }: CreateEditReviewProps) {
+  const { busID, editReviewId } = route.params;
+  const initialReview = editReviewId
+    ? useAppSelector(selectReviewById(editReviewId))
+    : undefined;
+  const dispatch = useDispatch();
+  const [newRating, setNewRating] = useState(initialReview?.rating ?? 0);
+  const [text, setText] = useState(initialReview?.comments ?? "");
+
+  const business = useAppSelector(selectBusinessById(busID));
+  const clientId = useAppSelector(selectUser)!.id;
+
+  const submitReview = async () => {
```

Comment:
might be worth putting in a try/catch since the dispatch depends on successful api calls? Just so that it doesn't go ahead with adding to redux even if it fails to do it in the db